### ISSUE TITLE
[sidekiq] Invalid value = NaN

### ISSUE THREAD

**Issue by Koki07** [mackerelio/mackerel-agent-plugins#593]
Hi.

I found that the value remains 0 when run the mackerel-sidekiq-plugin.
- [Maybe the same problem](https://twitter.com/mato_kata/status/1166727007113117697?s=20) (Japanese)

- my config
  - Rails v5.2.4.1
  - Sidekiq(ActiveJob) v5.2.7

Actually, the result of executing the error is as follows.
```
2020/02/21 18:40:04 Invalid value: key = sidekiq.ProcessedANDFailed.processed, value = NaN
2020/02/21 18:40:04 Invalid value: key = sidekiq.ProcessedANDFailed.failed, value = NaN
```

Thanks.

**Comment by lufia** [comment]
Hi, @Koki07 
We'd like to know your environment.

- Which version of check-plugins do you use?
- Which concrete numbers "prcessed" or "failed" from your Redis?
    - e.g: `redis-cli GET processed`
- Which concrete numbers ".last_diff.processed" or ".last_diff.failed" in your state file of the plugin?

The state file is parhaps located in a one of:

1. A specified path if there is `-tempfile` option
2. $MACKEREL_PLUGIN_WORKDIR/mackerel-plugin-sidekiq-xxx (xxx is a hash)
3. $TMPDIR/mackerel-plugin-sidekiq-xxx
4. /tmp/mackerel-plugin-sidekiq-xxx

**Comment by Koki07** [comment]
Hi @lufia 
Sorry for late reply.
See below for answers to the questions.

- mackerel-check-plugins.x86_64 v0.33.1
- redis concrete numbers
  - get "stat:failed:2020-02-21": 18
  - get "stat:processed:2020-02-21": 38
- state file concrete numbers
  - .last_diff.processed: 0
  - .last_diff.failed: 0

**Comment by tanatana** [comment]
Hi, @ Koki07

We found a bug in https://github.com/mackerelio/go-mackerel-plugin-helper.
We are going to fix the bug. After that, sidekiq plugin would work fine probably.

We update this issue if there is any progress.

Thank you!

**Comment by Koki07** [comment]
Hi, @tanatana 

Thank you for finding it out!
Has there been progress? I'm glad if you tell me.

**Comment by lufia** [comment]
Hi @Koki07.

We will update mackerel-plugin libraries by this week. We guesses that this update disappear the "Invalid value: key ... NaN" message.

However I think root cause of the error is the plugin runs same time. The first process writes the state file as `_lastTime=<now>`. In the same time, the second process will read the state. Because the libraries uses `(<currentValue> - <lastValue>) / (<currentTime> - <lastTime>)` for calculating the amount of increase, NaN occurs here if `<currentTime> - <lastTime>` is zero.

Could you know anything?

**Comment by lufia** [comment]
We have released new version of mackerel-agent-plugins v0.60.2, including fix for this.
